File: src/ObjectManager/Object.Tes/FilePacks/Records/345-GLOB.Global.cs
using OA.Core;
using System;

namespace OA.Tes.FilePacks.Records
{
    public class GLOBRecord : Record, IHaveEDID
    {
        public override string ToString() => $"GLOB: {EDID.Value}";
        public STRVField EDID { get; set; } // Global ID
        public BYTEField? FNAM; // Type of global (s, l, f)
        public FLTVField? FLTV; // Float data

        public override bool CreateField(UnityBinaryReader r, string type, uint dataSize)
        {
            switch (type)
            {
                case "NAME": EDID = new STRVField(r, dataSize); return true;
                case "FNAM": FNAM = new BYTEField(r, dataSize); return true;
                case "FLTV": FLTV = new FLTVField(r, dataSize); return true;
                default: return false;
            }
        }

        public override bool CreateField(UnityBinaryReader r, GameFormatId formatId, string type, uint dataSize)
        {
            switch (type)
            {
                case "EDID": EDID = new STRVField(r, dataSize); return true;
                case "FNAM": FNAM = new BYTEField(r, dataSize); return true;
                case "FLTV": FLTV = new FLTVField(r, dataSize); return true;
                default: return false;
            }
        }
    }
}

File: src.mac/ObjectManager/ObjectManager/Object.Tes/FilePacks/Records/345-GLOB.Global.swift
//
//  GLOBRecord.swift
//  ObjectManager
//
//  Created by Sky Morey on 5/28/18.
//  Copyright © 2018 Sky Morey. All rights reserved.
//

public class GLOBRecord: Record, IHaveEDID {
    public override var description: String { return "GLOB: \(EDID)" }
    public var EDID: STRVField = STRVField_empty // Editor ID
    public var FNAM: BYTEField? = nil // Type of global (s, l, f)
    public var FLTV: FLTVField? = nil // Float data

    override func createField(_ r: BinaryReader, for format: GameFormatId, type: String, dataSize: Int) -> Bool {
        switch type {
        case "EDID",
             "NAME": EDID = r.readSTRV(dataSize)
        case "FNAM": FNAM = r.readT(dataSize)
        case "FLTV": FLTV = r.readT(dataSize)
        default: return false
        }
        return true
    }
}
